server/src/main/java/org/schemarepo/server/JsonRenderer.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.schemarepo.server;
+
+import java.util.Properties;
+
+import javax.ws.rs.core.MediaType;
+
+import org.schemarepo.RepositoryUtil;
+import org.schemarepo.SchemaEntry;
+import org.schemarepo.Subject;
+import org.schemarepo.json.JsonUtil;
+
+/**
+ * Renders as JSON. The actual work is delegated to utility class {@link org.schemarepo.RepositoryUtil}
+ */
+public class JsonRenderer implements Renderer {
+
+  private final JsonUtil jsonUtil;
+
+  public JsonRenderer(JsonUtil jsonUtil) {
+    if (jsonUtil == null) {
+      throw new IllegalArgumentException("jsonUtil required");
+    }
+    this.jsonUtil = jsonUtil;
+  }
+
+  @Override
+  public String getMediaType() {
+    return MediaType.APPLICATION_JSON;
+  }
+
+  @Override
+  public String renderSubjects(Iterable<Subject> subjects) {
+    return jsonUtil.subjectsToJson(subjects);
+  }
+
+  @Override
+  public String renderSchemas(Iterable<SchemaEntry> schemaEntries) {
+    return jsonUtil.schemasToJson(schemaEntries);
+  }
+
+  @Override
+  public String renderSchemaEntry(SchemaEntry schemaEntry, boolean requestForLatest) {
+    return requestForLatest ? schemaEntry.toString() : schemaEntry.getSchema();
+  }
+
+  @Override
+  public String renderProperties(Properties props, String comment) {
+    // TODO: when the need arises, this may yield an actual JSON
```

Comment:
I guess there are a few aspects to this...

First of all, there are two notions of "default":

1. What the default response should be when no Accept header at all is specified. Here, I think we agree plain text is fine...
2. What the default response should be when an unsupported Accept header is specified. This is where we disagree, I think. I think it is safer to fail explicitly (via 415) than to make assumptions on a client's ability to process plain text (or some other default) when they explicitly asked for something else.

Secondly, what component should choose how to behave when a ```Renderer``` doesn't support a given operation? I believe it should be the ```RESTRepository``` that chooses what to do, not the ```Renderer```. Otherwise, there is code duplication and coupling of the ```Renderer``` with what should be a default and generic error handling code. If we decide to change what the function call should be for handling the plain text response, that should not require modifying the corresponding unimplemented operations in unrelated renderers....

Finally, regarding HTML-only endpoints, I didn't really mean that we should have that. I just meant that if we want to implement an HTML version of an endpoint that is currently just in plain text, you would need to add that operation to the ```Renderer``` interface, and it would remain unimplemented in the json one, thus further expanding the gap in functionality between the various renderers.